### ISSUE TITLE
[Wine] In-tree builds fail

### ISSUE THREAD

**Issue by mcneish1** [spack/spack#10056]
I'm trying to build Wine with spack.

Wine's build system attempts to link `[full spack staging path]/loader/l_intl.nls` to `loader/l_intl.nls`; unfortunately they're the same file. `configure` notices that it can't access the file anymore, so that step fails the build.

Doing `build_directory = 'build'` fixes this, (and I think deleting `WINE_CONFIG_SYMLINK(loader,loader,[l_intl.nls])` then re-running `autoreconf` does too but that might break other things).

Has anyone seen anything like this before? Is changing the build directory the usual solution for this, if so?

**Comment by mcneish1** [comment]
Related, should I PR this package? Wine requires a 32-bit compiler*, and it's not possible to build one of those with spack (unless you edit gcc/llvm's package files and _already have a 32-bit compiler_)

*there's also 64-bit Wine, but it's not compatible with most things someone would use Wine for

**Comment by mcneish1** [comment]
For posterity:
```
class Wine(AutotoolsPackage):
    """Wine (originally an acronym for "Wine Is Not an Emulator") is a
    compatibility layer capable of running Windows applications on several
    POSIX-compliant operating systems, such as Linux, macOS, & BSD."""

    homepage = "http://www.winehq.org"
    url      = "https://dl.winehq.org/wine/source/3.x/wine-3.21.tar.xz"

    version('3.21', sha256='a84cc06015df06e12c524213a98d879caa0d63f474b911cdd87f693fcfe2e0c0')

    build_directory = 'build'

    depends_on('libpng')
    depends_on('freetype')
    depends_on('libx11')
```
(note that with a crosstool-built gcc_i386, perl doesn't work with gdbm. berkeley-db works)

**Comment by adamjstewart** [comment]
> Has anyone seen anything like this before? Is changing the build directory the usual solution for this, if so?

Yep, we commonly switch to `build_directory = 'spack-build'` for this reason.

> Related, should I PR this package?

Yes, please! I'm not sure if Spack still supports 32-bit, but I'm sure someone could use it.

**Comment by mcneish1** [comment]
Hey Adam, unfortunately I haven't touched this in the last year and a half, and I don't have time to put this together nicely.

The recipe I posted above works for my purposes, but it isn't complete the 32-bit compiler requirement is difficult to deal with.

Here's a recipe (roughly the same age / completeness as the wine recipe) for crosstool-ng, to build said 32-bit compiler where needed:
```
from spack import *


class CrosstoolNg(AutotoolsPackage):
    homepage = "http://www.example.com"
    url      = "http://crosstool-ng.org/download/crosstool-ng/crosstool-ng-1.24.0.tar.xz"

    version('1.24.0', sha512='89b8794a4184ad4928750e29712ed4f194aa1d0b93768d67ff64f30c30f1b1e165647cafc6de94d68d3ef70e50446e544dad65aa36137511a32ee7a667dddfb4')

    depends_on('gperf')
    depends_on('help2man')
```

I used the ct-ng .config file here: https://gist.github.com/mcneish1/127eb433af49f1fab0f0e8a1aced55b7 though starting from scratch probably makes more sense (double check libc version and kernel header target, in particular).
